aws/table_aws_pricing_product.go

```diff
@@ -0,0 +1,274 @@
+package aws
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/aws/aws-sdk-go-v2/service/pricing"
+	"github.com/aws/aws-sdk-go-v2/service/pricing/types"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/turbot/steampipe-plugin-sdk/v4/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin/transform"
+	"strconv"
+	"time"
+
+	"github.com/turbot/steampipe-plugin-sdk/v4/plugin"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsPricingProduct(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "aws_pricing_product",
+		Description: "AWS Pricing Product",
+		List: &plugin.ListConfig{
+			Hydrate: listPricingProduct,
+			KeyColumns: []*plugin.KeyColumn{
+				{Name: "service_code", Require: plugin.Required},
+				{Name: "location_type", Require: plugin.Optional},
+				{Name: "region_code", Require: plugin.Optional},
+				{Name: "operating_system", Require: plugin.Optional},
+				{Name: "instance_type", Require: plugin.Optional},
+				{Name: "pre_installed_software", Require: plugin.Optional},
+				{Name: "capacity_status", Require: plugin.Optional},
+			},
+		},
+		Columns: awsDefaultColumns([]*plugin.Column{
+			{Name: "rate_code", Description: "A unique code for a product/ offer/ pricing-tier combination. The product and term combinations can have multiple price dimensions, such as a free tier, low-use tier, and high-use tier.", Type: proto.ColumnType_STRING, Transform: transform.FromField("Offer.PriceDimension.RateCode")},
+			{Name: "service_code", Description: "This identifies the specific AWS service to the customer as a unique short abbreviation.", Type: proto.ColumnType_STRING, Transform: transform.FromField("ServiceCode")},
+			{Name: "term", Description: "Whether your AWS usage is Reserved or On-Demand.", Type: proto.ColumnType_STRING, Transform: transform.FromField("Offer.Term")},
+			{Name: "purchase_option", Description: "How you chose to pay for this line item. Valid values are All Upfront, Partial Upfront, and No Upfront.", Type: proto.ColumnType_STRING, Transform: transform.FromField("Offer.TermAttributes.PurchaseOption")},
+			{Name: "lease_contract_length", Description: "The length of time that your RI is reserved for.", Type: proto.ColumnType_STRING, Transform: transform.FromField("Offer.TermAttributes.LeaseContractLength")},
+			{Name: "description", Description: "Description for a product / offer / pricing-tier combination", Type: proto.ColumnType_STRING, Transform: transform.FromField("Offer.PriceDimension.Description")},
+			{Name: "begin_range", Type: proto.ColumnType_STRING, Transform: transform.FromField("Offer.PriceDimension.BeginRange")},
+			{Name: "end_range", Type: proto.ColumnType_STRING, Transform: transform.FromField("Offer.PriceDimension.EndRange")},
```

Comment:
Please add the column description for the `begin_range` and the `end_range` columns.